### ISSUE TITLE
Build Training Pipeline

### ISSUE THREAD

**Issue by joseph-zhong** [joseph-zhong/LipReading#2]
We can probably take inspiration from deep-speech using CTC loss

- pytorch training loop: https://github.com/SeanNaren/deepspeech.pytorch/blob/master/train.py
- pytorch transcript caption label dataloader:
https://github.com/SeanNaren/deepspeech.pytorch/blob/master/data/data_loader.py#L161

- [x] training loop refactor
- [x] variable seq-len Dataloader
